final-progII/src/presentation/Screen.java

```diff
@@ -1382,6 +1383,32 @@ public void actionPerformed(ActionEvent e) {
 
 			}
 		});
+
+		submitCreateSubject__button.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+
+				String code = codeCreateSubject__textField.getText();
+				String name = nameCreateSubject__textField.getText();
+				String orientation = orientationCreateSubject__textField.getText();
+				String generation = generationCreateSubject__textField.getText();
+				int teacherCI = Integer.valueOf(teacherCreateSubject__textField.getText());
+
+				try {
+
+					Subject subject = new Subject(code, name, Orientation.valueOf(orientation),
+							Generation.valueOf(generation), new Teacher(teacherCI));
```

Comment:
How it works that of `new Teacher(teacherCI),` why don't you search a teacher with the ci and if its exits create a subject.